src/observers/http1/decoder.h

```diff
@@ -0,0 +1,141 @@
+#ifndef http1_observer_decoder_H
```

Comment:
Regarding the location of the decoder code, where it is now under the observer does not enforce any kind of separation.  Maybe that's a desired coupling?  If it stays here, it's easy to see how it would grow mutual dependencies with the other observer code.

Reply:
> Isn't the convention to make these all caps?

I don't believe the convention is all caps but I'll update the ifdef to be consistent with other existing header files.

Reply:
> Regarding the location of the decoder code, where it is now under the observer does not enforce any kind of separation. Maybe that's a desired coupling? If it stays here, it's easy to see how it would grow mutual dependencies with the other observer code.

That's a good point - I'll follow up with a separate PR to re-organize the file layout to make the decoupling clearer.  Stay tuned.